articles/151_roslaunch_xml.md

```diff
@@ -0,0 +1,744 @@
+              <xs:annotation>
+                <xs:documentation xml:lang="en">
+                  ROS node output goes to a log file.
+                </xs:documentation>
+              </xs:annotation>
+            </xs:enumeration>
+            <xs:enumeration value="screen">
+              <xs:annotation>
+                <xs:documentation xml:lang="en">
+                  ROS node output goes to stdout.
+                </xs:documentation>
+              </xs:annotation>
+            </xs:enumeration>
+          </xs:restriction>
+        </xs:simpleType>
+      </xs:attribute>
+      <xs:attribute name="if" type="xs:string" use="optional">
+        <xs:annotation>
+          <xs:documentation xml:lang="en">
+            A predicate to condition ROS node launch i.e. only do
+            so if the predicate evaluates to a truthy value.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:attribute>
+      <xs:attribute name="unless" type="xs:string" use="optional">
+        <xs:annotation>
+          <xs:documentation xml:lang="en">
+            A predicate to condition ROS node launch i.e. do so unless
+            the predicate evaluates to a truthy value.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:attribute>
+  </xs:element>
+</xs:schema> 
+```
+
+### Tags Semantics
+
+#### All Tags
+
+Every tag's execution can be conditioned on a boolean predicate via `if` or `unless` attributes.
+
+#### <launch> Tag
+
+Root tag of any launch file.
+
+#### <include> Tag
+
+##### Description
+
+The <include> tag allows for bringing a launch file description into another, enabling re-use of hierarchical launch layouts.
+The included launch file description has its own scope for launch configurations.
+The included launch file description is not necessarily written in this format nor a declarative one (i.e. a programmatic description).
+To avoid ROS name clashes, included nodes can be namespaced via the `ns` attribute.
+
+##### Examples
+
+```xml
+<include file="/opt/my_launch_file.py" ns="/my_ns"/>
+<include file="/opt/my_other_launch_file.xml">
+  <arg name="some_argument" value="dummy_value"/>
+</include>
+```
+
+#### <group> Tag
+
+##### Description
+
+The <group> tag allows for launch actions' grouping as well as optional launch file configuration scoping.
+
+##### Examples
+
+```xml
+<group ns="/dummy_group" scoped="true">
+  <node package="a_ros_package" name="dummy0" executable="dummy_node"/>
+  <node package="a_ros_package" name="dummy1" executable="dummy_node"/>
+</group>
+```
+
+#### <arg> Tag
+
+##### Description
+
+The <arg> tag allows for launch file configuration via the command-line or when including it via an <include> tag.
+Arguments are limited to the scope of their definition and thus, have to be explictly passed to included files if any.
+
+##### Examples
+
+```xml
+<arg name="publish_frequency" default="10"/>
+<arg name="output_path" doc="Output path for some processing pipeline"/>
```

Comment:
I think `description` is the term used in launch, rather than `doc`.

Reply:
That's correct. Changed.